test/helpers/hosts_and_hostgroups_helper_test.rb

```diff
@@ -244,4 +259,8 @@ def setup
     @hostgroup.medium_id = id
     assert_nil kickstart_repository_id(::Host.new, :selected_host_group => @hostgroup)
   end
+
+  test 'repository_url must render the right path based on host configuration' do
+    assert_equal "http://#{@host.content_source.hostname}/#{@host.organization.label}/#{@host.lifecycle_environment}/", repository_url('/')
```

Comment:
An example of how the structure of url looks like is something like:

```
http://katello.example.com/pulp/repos/Default_Organization/Library/custom/zoo/zoo/
```

So we need to add `pulp/repos` at the beginning,

Also, I recommend to test with something more advanced, than with just '/' as argument in  `repository_url('/')`